packages/glimmer-runtime/lib/dom.ts

```diff
@@ -153,6 +153,9 @@ class DOMHelper {
   }
 
   insertBefore(element: Element, node: Node, reference: Node) {
+    if (reference && element.children.length === 0) {
```

Comment:
This shouldn't be changed to accepting DocumentFragment, it should be resolved in the caller.